### ISSUE TITLE
[SUGGEST] Rename Pokemons

### ISSUE THREAD

**Issue by 30er** [Spegeli/PokemoGoBot-GottaCatchEmAll#216]
Would it be Possible to Auto rename Pokemons with the PerfectionRate?

f.e.
[PERFECTION] #ID
[12.34%] #22

So u know how perfect they are when u open the real app.
If you want to do it advanced you could do also a setting where we can choose our naming with placeholders.

f.e.
<setting>#ID# #CP# #HP# #PF# #LVL#....</setting>

I know there is a limit of length ;) but shouldnt be hard to trim the resulting nick.

Would be awesome ;)

**Comment by Bakeneko** [comment]
I wrote an auto-rename functionality (IV_PokemonId template) but it seems too easy to spot.
I think i'm gonna randomize IV and Id. Thoughts?

**Comment by AsmoAeon** [comment]
I rename them according to their IV appending Z0 O1 T2 H3 F4 I5 S6 E7 G8 N9  to their names (zero0=00 one1=11 N9=Ninety nine, N5=Ninety five, you get the idea). I can see their CP and HP anyway.

Btw niantic will change the protocol soon to be lighter to their servers.

**Comment by 30er** [comment]
i dont think that its "easy" to spot... i did it allready myself before i used the bot... but it took ages :D.
as i suggested before,,, make it configurable fia settings...
so offer users some way to customize there naming

f.e.
user1:
#ID# #PF#

user2:
#PF# #ID#

user3:
SomeText[#ID#-#PF#]

so its configurable...

furthermore it would be nice if you can autoadd 100% creeps to Favorites (or make a setting for Percentage for auto favorite)

Greez

**Comment by Bakeneko** [comment]
The code i wrote filter favorites and pokemons at gym, and use a template. You can use {IV}, {CP}, {ID}.
By default the template is "{IV}_{ID}", for example : a 100% Dragonair is renamed 100_Dragonai (12 chars max).

I just need to fix some shit with this git repo whitespaces and submit a PR to @Spegeli

**Comment by 30er** [comment]
can u change {ID} to real id
and add another placeholde {Name} for the name?
